### ISSUE TITLE
`no-magic-numbers` doesn't catch assignment operators

### ISSUE THREAD

**Issue by freejosh** [eslint/eslint#4400]
Using ESLint 1.9.0:

Input:
```js
/*eslint no-magic-numbers: 2*/
var a = 5;
a += 5;
a = a + 5;
```

Result:
```
  4:13  error  No magic number: 5  no-magic-numbers

✖ 1 problem (1 error, 0 warnings)
```

I'd expect to also get an error for line 3.

**Comment by alberto** [comment]
Seems reasonable to me. @eslint/eslint-team opinions?

**Comment by ilyavolodin** [comment]
I think this is not an enhancement but a bug, and I think we should fix this.

**Comment by platinumazure** [comment]
:+1: to @ilyavolodin on both counts.
